### ISSUE TITLE
Copy method for DataQualityDict doesn't deepcopy

### ISSUE THREAD

**Issue by tjma12** [gwpy/gwpy#710]
@duncanmmacleod It looks like the `copy()` method for `DataQualityDict` isn't working as intended. If I make a `DataQualityDict`, copy it, and modify the underlying segment lists, both dictionaries are modified.

Example from ipython below:
```python
In [1]: from gwpy.segments import DataQualityFlag, DataQualityDict, Segment, SegmentList

In [2]: seg1 = Segment(5,10)

In [3]: seg2 = Segment(12,15)

In [4]: seg3 = Segment(4,8)

In [5]: seg4 = Segment(11,14)

In [6]: seglist1 = SegmentList([seg1,seg2])

In [7]: seglist2 = SegmentList([seg3,seg4])

In [8]: flag1 = DataQualityFlag('TEST', active=seglist1, known=seglist1)

In [9]: flag2 = DataQualityFlag('TEST', active=seglist2, known=seglist2)

In [10]: dict1 = DataQualityDict({'f1': flag1})

In [11]: dict2 = DataQualityDict({'f1': flag2})

In [12]: cdict1 = dict1.copy()

In [13]: dict1
Out[13]: 
DataQualityDict([('f1', <DataQualityFlag('TEST',
                 known=[[5 ... 10)
                        [12 ... 15)],
                 active=[[5 ... 10)
                         [12 ... 15)],
                 description=None)>)])

In [14]: cdict1
Out[14]: 
DataQualityDict([('f1', <DataQualityFlag('TEST',
                 known=[[5 ... 10)
                        [12 ... 15)],
                 active=[[5 ... 10)
                         [12 ... 15)],
                 description=None)>)])

In [15]: flag1
Out[15]: 
<DataQualityFlag('TEST',
                 known=[[5 ... 10)
                        [12 ... 15)],
                 active=[[5 ... 10)
                         [12 ... 15)],
                 description=None)>

In [16]: flag2
Out[16]: 
<DataQualityFlag('TEST',
                 known=[[4 ... 8)
                        [11 ... 14)],
                 active=[[4 ... 8)
                         [11 ... 14)],
                 description=None)>

In [17]: flag1 ^= flag2

In [18]: flag1
Out[18]: 
<DataQualityFlag('TEST',
                 known=[[4 ... 5)
                        [8 ... 10)
                        [11 ... 12)
                        [14 ... 15)],
                 active=[[4 ... 5)
                         [8 ... 10)
                         [11 ... 12)
                         [14 ... 15)],
                 description=None)>

In [19]: dict1
Out[19]: 
DataQualityDict([('f1', <DataQualityFlag('TEST',
                 known=[[4 ... 5)
                        [8 ... 10)
                        [11 ... 12)
                        [14 ... 15)],
                 active=[[4 ... 5)
                         [8 ... 10)
                         [11 ... 12)
                         [14 ... 15)],
                 description=None)>)])

In [20]: cdict1
Out[20]: 
DataQualityDict([('f1', <DataQualityFlag('TEST',
                 known=[[4 ... 5)
                        [8 ... 10)
                        [11 ... 12)
                        [14 ... 15)],
                 active=[[4 ... 5)
                         [8 ... 10)
                         [11 ... 12)
                         [14 ... 15)],
                 description=None)>)])
```

**Comment by duncanmmacleod** [comment]
@tjma12, it turns out that `DataQualityDict` *does not* overload `.copy()`, so this method (I think) is working exactly [as advertised](https://gwpy.github.io/docs/stable/api/gwpy.segments.DataQualityDict.html#gwpy.segments.DataQualityDict.copy):

```
$ python -c "from gwpy.segments import DataQualityDict; help(DataQualityDict.copy)"
Help on method copy in module collections:

copy(self) unbound gwpy.segments.flag.DataQualityDict method
    od.copy() -> a shallow copy of od
```

That said, I am happy to implement an overloaded version with a keyword `deep` that will perform a full `deepcopy` of all values.

**Comment by tjma12** [comment]
@duncanmmacleod Ah, I see. I assumed it was intended to be a `deepcopy` based on the methods that are supposed to return a new `DataQualityDict` rather than an in-place modification.

**Comment by duncanmmacleod** [comment]
@tjma12, do you fancy taking a shot at this one yourself?

**Comment by tjma12** [comment]
@duncanmmacleod Sure, I'll see if I can get this and #704 all pulled together

**Comment by duncanmmacleod** [comment]
This was closed by #704.
